File: apps/s-2-admin/src/api/availability/AvailabilityWhereInput.ts
import { IntFilter } from "../../util/IntFilter";
import { UserWhereUniqueInput } from "../user/UserWhereUniqueInput";
import { EventTypeWhereUniqueInput } from "../eventType/EventTypeWhereUniqueInput";
import { DateTimeFilter } from "../../util/DateTimeFilter";
import { DateTimeNullableFilter } from "../../util/DateTimeNullableFilter";
import { ScheduleWhereUniqueInput } from "../schedule/ScheduleWhereUniqueInput";

export type AvailabilityWhereInput = {
  id?: IntFilter;
  user?: UserWhereUniqueInput;
  eventType?: EventTypeWhereUniqueInput;
  days?: IntFilter;
  startTime?: DateTimeFilter;
  endTime?: DateTimeFilter;
  date?: DateTimeNullableFilter;
  schedule?: ScheduleWhereUniqueInput;
};

File: apps/cactus-demo/src/APIs/Availability/Dtos/AvailabilityWhereInput.cs
namespace CactusDemo.APIs.Dtos;

public class AvailabilityWhereInput
{
    public int? Id { get; set; }

    public int? Days { get; set; }

    public DateTime? StartTime { get; set; }

    public DateTime? EndTime { get; set; }

    public DateTime? Date { get; set; }

    public int? EventType { get; set; }

    public int? Schedule { get; set; }

    public int? Users { get; set; }
}
